apps/web/src/pages/index.tsx

```diff
@@ -1,5 +1,42 @@
 import React from "react"
-import { useSession } from "next-auth/react"
+import { useSession as _useSession } from "next-auth/react"
+import { trpc } from "@/utils/trpc"
+
+export const useSession = (): {
+  isLoading: boolean
+  user: {
+    id: string
+    email: string
+    name: string
+    image: string
+    studyYear: number
+  }
+} => {
+  const { data: session, status } = _useSession()
+  const { data, isLoading: dbCallLoading } = trpc.user.get.useQuery(session?.user.id ?? "", {
+    enabled: Boolean(session?.user.id),
+  })
+
+  // Initialize user with default values
+  const defaultUser = {
+    id: "",
+    email: "",
+    name: "",
+    image: "",
+    studyYear: 0,
+  }
+
+  return {
+    isLoading: dbCallLoading || status === "loading",
+    user: {
```

Comment:
I believe this is way better typed as User|null when it's loading.

Reply:
We shouldn't return fake/empty data when stuff is loading, we should return a value that is indicating something is loading/not present